File: frontend/src/microservices/operations/siteadmin/objects/SiteAdminManagerSessionPlannerResponse.ts
// 本文件定义 siteadmin 子域自己的管理员会话返回体。
// 它对应 RegisterSiteAdminPlanner / UpdateSiteAdminManagerProfilePlanner 的返回数据，不再借用 overall 子域。
export type SiteAdminManagerSessionPlannerResponse = {
  managerId: string
  managerType: string
  email: string
  displayName: string
  status: string
  scopeId: string
  logoAssetPath?: string | null
  createdAt: string
}

export const siteAdminManagerSessionPlannerResponseFromJson = (json: string): SiteAdminManagerSessionPlannerResponse =>
  JSON.parse(json) as SiteAdminManagerSessionPlannerResponse

export const siteAdminManagerSessionPlannerResponseToJson = (value: SiteAdminManagerSessionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/siteadmin/objects/SiteAdminManagerSessionPlannerResponse.scala
// 这个文件定义 siteadmin 子域自己的管理员会话返回体。
// 它只对应站点管理员的注册与资料更新流程，不再借用 operations/overall 的共享壳。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SiteAdminManagerSessionPlannerResponse(
    managerId: String,
    managerType: String,
    email: String,
    displayName: String,
    status: String,
    scopeId: String,
    logoAssetPath: Option[String],
    createdAt: String
)
object SiteAdminManagerSessionPlannerResponse:
  given sourceEncoder: Encoder[SiteAdminManagerSessionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[SiteAdminManagerSessionPlannerResponse] = deriveDecoder
